test_suite.py

```diff
@@ -207,10 +227,13 @@ class TestStreamingStdin:
     without waiting for all lines to arrive
     Note: this test uses Docker CLI instead of the Python Docker SDK (implemented in the
     `docker_runner` fixture) since SDK doesn't easily allow writing to a container's stdin"""
-    def test_stdin(self, docker_image):
-        # Subprocess constructor that runs the py script in a docker container and waits for input
+    def test_stdin(self, docker_image, language):
+        # Need 2 separate strings for subprocess command
+        command, script_name = language.script("streaming_stdin").split()
```

Comment:
This test needs the command to call the script formatted a bit differently, so just splitting the output of the script method. Other option would be to add the script as a parameter to the class, which would be cleaner, but only used in this one test

Reply:
Hmmmm... yeah

Splitting seems ok-ish, but it might bite us later. Like, what if we have a`Blublang` langage that gets run like like:

```sh
blug streaming_stdin.blug --use="stdout stdin"
```
...that would get split into:
```python
['blug', 'streaming_stdin.blug', '--use="stdout', 'stdin"']
```

So, I'd say we either:
1. Go with the `.split()` technique for now until we need to change it. If we do that, we'd probably split then splat  like this. 
    ```python
    command_parts = language.script("streaming_stdin").split()
    ['docker', 'run', '-i', docker_image, *command_parts]
    ```
    This will handle *any* number of parts

2. Change the `.script(test_name)` method to return a list of the parts already (e.g. `["python", "null_char.py"]` instead of `"python null_char.py"`). E.g.
    ```python
    def script(self, test_name):
      return ["python", f"{test_name}.py"]
    ```

3. Make a Language method called something like `.script_command_parts(test_name)` that returns the command already split correctly.
    ```python
    class Python:
      # ...
      def script_command_parts(self, test_name):
        return ["python", f"{test_name}.py"]

      def script(self, test_name):
        # no need to repeat ourselves if we have `script_command_parts`
        return " ".join(self.script_command_parts(test_name))

    # ...then use it like this in a test:
    script = subprocess.Popen(
      ['docker', 'run', '-i', docker_image, *language.script_command_parts("streaming_stdin")],
      # ...
    )
    ```

Out of these, I'm partial to number 3, but I'm interested in what you think.